extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocationKey.java

```diff
@@ -48,4 +76,76 @@ public String getImplementationName() {
     public ParquetInstructions readInstructions() {
         return readInstructions;
     }
+
+    /**
+     * Precedence-wise this implementation compares {@code order}, then {@code dataSequenceNumber}, then
+     * {@code fileSequenceNumber}, then {@code pos}, then applies a {@link PartitionsComparator} to {@code partitions}
+     * and finally {@code uri}.
+     *
+     * @inheritDoc
+     */
+    @Override
+    public int compareTo(@NotNull final TableLocationKey other) {
+        if (other instanceof IcebergTableParquetLocationKey) {
+            final IcebergTableParquetLocationKey otherTyped = (IcebergTableParquetLocationKey) other;
+            int comparisonResult;
+            if ((comparisonResult = Integer.compare(order, otherTyped.order)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(dataSequenceNumber, otherTyped.dataSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(fileSequenceNumber, otherTyped.fileSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(dataFilePos, otherTyped.dataFilePos)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = Long.compare(manifestSequenceNumber, otherTyped.manifestSequenceNumber)) != 0) {
+                return comparisonResult;
+            }
+            if ((comparisonResult = PartitionsComparator.INSTANCE.compare(partitions, otherTyped.partitions)) != 0) {
+                return comparisonResult;
+            }
+            return uri.compareTo(otherTyped.uri);
+        }
+        throw new ClassCastException("Cannot compare " + getClass() + " to " + other.getClass());
+    }
+
+    @Override
+    public boolean equals(@Nullable final Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof IcebergTableParquetLocationKey)) {
+            return false;
+        }
+        final IcebergTableParquetLocationKey otherTyped = (IcebergTableParquetLocationKey) other;
+        return dataSequenceNumber == otherTyped.dataSequenceNumber
+                && fileSequenceNumber == otherTyped.fileSequenceNumber
+                && dataFilePos == otherTyped.dataFilePos
+                && manifestSequenceNumber == otherTyped.manifestSequenceNumber
+                && super.equals(otherTyped);
```

Comment:
I actually think this might be a bad change. We need to decide how adding a URI at different sequence numbers should be handled.
I could see this creating a spurious remove and add pair.

Reply:
Keeping it this way since we saw cases where different data files had the same URIs.
Added a comment explaining the same.